test-it/ViaTests/QueueRefNumEnqueueFrontBug.via

```diff
@@ -0,0 +1,44 @@
+define(TestProgram dv(.VirtualInstrument (
+    c(
+        e(.QueueRefNum<.Double> q)
+        e(.Double y)
+        e(.Int32 maxsize)
+        e(.Int32 ne)
+        e(.Boolean timedOut)
+        e(.Boolean overflowed)
+        e(.Boolean badAliasingTestPass)
+        e(a(.Double *) remainingElts)
+        e(.ErrorCluster err)
+    )
+
+    clump (
+        ObtainQueue(q * * * * err)
+        Enqueue(q 0.0 1000 timedOut err)
+        Printf("Enqueue with empty unbounded queue\n")
+        GetQueueStatus(q true maxsize * * * ne remainingElts err)

```

Comment:
I think we should print err (at least, err.status and err.code).

Reply:
Even with queue of fixed size, following case fail where last Deque return the incorrect value.
1) Enqueue, Deque, EnqueueFront, Dequeue
2) EnqueueFront, Deque, EnqueueFront, Dequeue
Please add more tests including the above. I would also use "PeekQueue" before Dequeue.  Adding More tests in VI test suite are acceptable. However, we should make sure those all pass for the proposed fix.